platform/web-security/src/main/java/org/motechproject/security/email/impl/EmailSenderImpl.java

```diff
@@ -1,90 +1,143 @@
 @Service
 public class EmailSenderImpl implements EmailSender {
-    private static final Logger LOGGER = LoggerFactory.getLogger(EmailSenderImpl.class);
 
-    private static final String RESET_MAIL_TEMPLATE = "/mail/resetMail.vm";
-    private static final String ONE_TIME_TOKEN_TEMPLATE = "/mail/oneTimeTokenMail.vm";
-    private static final String RECOVERY_SUBJECT = "Motech Password Recovery";
-    private static final String ONE_TIME_TOKEN_SUBJECT = "Motech One Time Token For Admin User";
-    private static final String LOGIN_INFORMATION_TEMPLATE = "/mail/loginInfo.vm";
-    private static final String LOGIN_INFORMATION_SUBJECT = "Motech Login Information";
-    private static final String EMAIL_PARAM_FROM_ADDRESS = "fromAddress";
-    private static final String EMAIL_PARAM_TO_ADDRESS = "toAddress";
-    private static final String EMAIL_PARAM_MESSAGE = "message";
-    private static final String EMAIL_PARAM_SUBJECT = "subject";
-    private static final String TEMPLATE_PARAM_LINK = "link";
-    private static final String TEMPLATE_PARAM_USERNAME = "user";
-    private static final String TEMPLATE_PARAM_MESSAGES = "messages";
-    private static final String TEMPLATE_PARAM_LOCALE = "locale";
-    private static final String RESET_PATH = "reset";
-    private static final String ONE_TIME_TOKEN_PATH = "onetimetoken";
+    private static final Logger LOGGER = LoggerFactory.getLogger(EmailSenderImpl.class);
 
     private EventRelay eventRelay;
-    private VelocityEngine velocityEngine;
+    private VelocityTemplateParser templateParser;
     private SettingsFacade settingsFacade;
     private ResourceBundleMessageSource messageSource;
 
     @Override
     public void sendRecoveryEmail(final PasswordRecovery recovery) {
-        LOGGER.info("Sending recovery email");
 
-        Map<String, Object> model = templateParams(recovery);
-        String text = VelocityEngineUtils.mergeTemplateIntoString(velocityEngine, RESET_MAIL_TEMPLATE, model);
+        LOGGER.info("Sending recovery e-mail");
+
+        try {
+
+            String text = templateParser.mergeTemplateIntoString(RESET_MAIL_TEMPLATE,
+                    passRecoveryTemplateParams(recovery));
+
+            String subject = messageSource.getMessage(RECOVERY_MESSAGE_SUBJECT, null, recovery.getLocale());
 
-        sendEmail(recovery.getEmail(), text, RECOVERY_SUBJECT);
+            sendEmail(recovery.getEmail(), text, subject);
+
+        } catch (VelocityTemplateParsingException e) {
+            LOGGER.error("Couldn't send recovery e-mail", e);
+        }
     }
 
     @Override
     public void sendOneTimeToken(final PasswordRecovery recovery) {
+
         LOGGER.info("Sending one time token");
 
-        Map<String, Object> model = templateParams(recovery);
-        String text = VelocityEngineUtils.mergeTemplateIntoString(velocityEngine, ONE_TIME_TOKEN_TEMPLATE, model);
+        try {
+
+            String text = templateParser.mergeTemplateIntoString(ONE_TIME_TOKEN_TEMPLATE,
```

Comment:
I am getting the following:

<html>
	<body>
		<strong>Hello
 motech, </strong>
     	<p>Password for account motech will expire in null days.</p>
		<p>Last password change date :
 $lastPasswordChange</p>
		<p>Password expiration date:
 $expirationDate</p>
		<p>To change your password please visit the link below and navigate to the password change page.
</p>
		<p><a href="$serverUrl">$serverUrl</a><p>
	</body>
</html>